state-chain/amm-math/src/lib.rs

```diff
@@ -109,16 +109,16 @@ pub fn output_amount_ceil(input: Amount, price: Price) -> Amount {
 	mul_div_ceil(input, price, U256::one() << PRICE_FRACTIONAL_BITS)
 }
 
-pub fn price_from_usd_fine_amount(fine_amount: Amount) -> Price {
-	Price::from(fine_amount) << PRICE_FRACTIONAL_BITS
+pub fn price_from_usd_fine_amount(price_usd: AssetAmount) -> Price {
```

Comment:
`AssetAmount` is misleading because we normally use it for fine amounts. As dicussed, it is OK to change it back to u32 as long as we explain that we do it for extra type safety.